Sources/PackageModel/Manifest/PackageDependencyDescription.swift

```diff
@@ -11,149 +11,189 @@
 import TSCBasic
 
 /// Represents a package dependency.
-public enum PackageDependencyDescription: Equatable {
-
-     public struct Local: Equatable, Codable {
+public enum PackageDependency: Equatable {
+    public struct FileSystem: Equatable, Encodable {
         public let identity: PackageIdentity
         public let name: String?
         public let path: AbsolutePath
         public let productFilter: ProductFilter
     }
 
-    public struct SourceControlRepository: Equatable, Codable {
+    public struct SourceControl: Equatable, Encodable {
         public let identity: PackageIdentity
         public let name: String?
         public let location: String
         public let requirement: Requirement
         public let productFilter: ProductFilter
-    }
-
-    case local(Local)
-    case scm(SourceControlRepository)
-    //case registry(data: Registry) // for future
-
-    /// The dependency requirement.
-    public enum Requirement: Equatable, Hashable {
-        case exact(Version)
-        case range(Range<Version>)
-        case revision(String)
-        case branch(String)
 
-        public static func upToNextMajor(from version: TSCUtility.Version) -> Requirement {
-            return .range(version..<Version(version.major + 1, 0, 0))
+        /// The dependency requirement.
+        public enum Requirement: Equatable, Hashable {
+            case exact(Version)
+            case range(Range<Version>)
+            case revision(String)
+            case branch(String)
         }
+    }
 
-        public static func upToNextMinor(from version: TSCUtility.Version) -> Requirement {
-            return .range(version..<Version(version.major, version.minor + 1, 0))
+    public struct Registry: Equatable, Encodable {
+        public let identity: PackageIdentity
+        public let requirement: Requirement
+        public let productFilter: ProductFilter
+
+        /// The dependency requirement.
+        public enum Requirement: Equatable, Hashable {
+            case exact(Version)
+            case range(Range<Version>)
         }
     }
 
+    case fileSystem(FileSystem)
+    case sourceControl(SourceControl)
+    case registry(Registry)
```

Comment:
@neonichu are you happy with these names? note `local` -> `fileSystem` and `scm` -> `sourceControl`

Reply:
The names seem good, and in general, this PR is an improvement.  I'm a bit concerned about the absolute path being stored for the path, which isn't easily round-trippable when generating the source for a package manifest, but it looks as if that was already the case.  That might be a bug that we have today (I think I've seen the same thing when describing a package, i.e. that the absolute path is emitted rather than the one that had been specified).  Since it isn't new in this PR, that can be fixed later.